Code/Framework/AzQtComponents/AzQtComponents/Components/FilteredSearchWidget.cpp

```diff
@@ -233,6 +235,44 @@ namespace AzQtComponents
         QMenu::showEvent(e);
     }
 
+    bool SearchTypeSelector::eventFilter(QObject* obj, QEvent* event)
+    {
+        // Focus the first visible item if pressing down from the search field
+        if (obj == m_searchField && event->type() == QEvent::KeyPress)
+        {
+            auto* keyEvent = static_cast<QKeyEvent*>(event);
+            if (keyEvent->key() == Qt::Key_Down && keyEvent->modifiers() == Qt::NoModifier)
+            {
+                if (const QModelIndex firstVisibleIdx = m_tree->indexAt({ 0, 0 });
+                    firstVisibleIdx.isValid())
+                {
+                    m_tree->setFocus();
+                    m_tree->setCurrentIndex(firstVisibleIdx);
+                    return true;
+                }
+            }
+        }
+
+        // Focus the search field when pressing up if the first item is selected
+        if (obj == m_tree && event->type() == QEvent::KeyPress)
```

Comment:
nit: Things could be reduced if the common checks for the event type, modifier state, and key event cast are moved to an outer if or switch statement. The check for the up and down keys could also be converted to a switch statement since both cases are mutually exclusive.

Reply:
Hey,
I took a look again and I moved the cast outside:
```cpp
    bool SearchTypeSelector::eventFilter(QObject* obj, QEvent* event)
    {
        if (event->type() == QEvent::KeyPress)
        {
            const auto* keyEvent = static_cast<QKeyEvent*>(event);

            // Focus the first visible item if pressing down from the search field
            if (obj == m_searchField &&
                keyEvent->key() == Qt::Key_Down && keyEvent->modifiers() == Qt::NoModifier)
            {
                if (const QModelIndex firstVisibleIdx = m_tree->indexAt({ 0, 0 }); firstVisibleIdx.isValid())
                {
                    m_tree->setFocus();
                    m_tree->setCurrentIndex(firstVisibleIdx);
                    return true;
                }
            }

            // Focus the search field when pressing up if the first item is selected
            if (obj == m_tree &&
                keyEvent->key() == Qt::Key_Up && keyEvent->modifiers() == Qt::NoModifier)
            {
                const QModelIndex firstItem = m_tree->model()->index(0, 0);
                const bool isFirstSelected = m_tree->selectionModel()->selectedIndexes().contains(firstItem);
                if (isFirstSelected)
                {
                    m_searchField->setFocus();
                    m_tree->clearSelection();
                    return true;
                }
            }
        }

        return QMenu::eventFilter(obj, event);
    }
```

I could maybe also take the modifier check outside but I don't think it would be for much benefit. It would be another if level. Also, in the above version both characteristics of the keystroke (key and modifier) are checked together in one line making it easy to see on what type of keystroke are we reacting.

Reply:
@gadams3 Please let me know if you think that's better. I'll do the change if yes.